doc/source/serve/llm/user-guides/model-loading.md

```diff
@@ -249,7 +249,143 @@ Use EC2 instance profiles or EKS service accounts with appropriate S3 read permi
+
+```python
+llm_config = LLMConfig(
+    ...
+    model_loading_config={
+        "model_id": "llama",
+        "model_source": "s3://your-bucket/Meta-Llama-3-8B-Instruct",
+    },
+    engine_kwargs={
+        "tensor_parallel_size": 1,
+        "load_format": "runai_streamer",
+    },
+    ...
+)
+```
+
+### Model Sharding
+Modern LLM model sizes often outgrow the memory capacity of a single GPU, requiring the use of tensor parallelism to split computation across multiple devices. In this paradigm, only a subset of weights are stored on each GPU, and model sharding ensures that each device only loads the relevant portion of the model. By presharding the model files in advance, we can reduce load times significantly, since GPUs avoid loading unneeded weights. vLLM provides a utility script for this purpose: [save_sharded_state.py](https://github.com/vllm-project/vllm/blob/main/examples/offline_inference/save_sharded_state.py).
+
+Once the sharded weights have been saved, upload them to S3 and use RunAI streamer with a new flag to load the sharded weights
+
+```python
+llm_config = LLMConfig(
+    ...
+    engine_kwargs={
+        "tensor_parallel_size": 4,
+        "load_format": "runai_streamer_sharded",
+    },
+    ...
+)
+```
+
+### Torch Compile Cache
+Torch.compile is a critical stage during setting up a model for inference, and it incurs some latency during initialization. This can be mitigated by keeping a torch compile cache, which is automatically generated by vLLM. To retrieve the torch compile cache, run vLLM and look for a log like below:
+```
+(RayWorkerWrapper pid=126782) INFO 10-15 11:57:04 [backends.py:608] Using cache directory: /home/ray/.cache/vllm/torch_compile_cache/131ee5c6d9/rank_1_0/backbone for vLLM's torch.compile
+```
+
+In this example the cache folder is located at `/home/ray/.cache/vllm/torch_compile_cache/131ee5c6d9`.
+Simply upload this file to your S3 bucket. The cache folder can now be retrieved at startup. We provide a custom utility to download the compile cache from cloud storage. Simply specify the `CloudDownloader` callback in `LLMConfig` and supply the relevant arguments. Make sure to set the `cache_dir` in compilation_config correctly. 
+
+```python
+llm_config = LLMConfig(
+    ...
+    callback_config={
+        "callback_class": "ray.llm._internal.common.callbacks.cloud_downloader.CloudDownloader",
+        "callback_kwargs": {"paths": [("s3://samplebucket/llama-3-8b-cache", "/home/ray/.cache/vllm/torch_compile_cache/llama-3-8b-cache")]},
+    },
+    engine_kwargs={
+        "tensor_parallel_size": 1,
+        "compilation_config": {
+            "cache_dir": "/home/ray/.cache/vllm/torch_compile_cache/llama-3-8b-cache",
+        }
+    },
+    ...
+)
+```
+
```

Comment:
add a note here that says CloudDownloader is a callback that is in not public yet. We plan to make it public after stabilizing the API and based on user feedback.